src/textual/_node_list.py

```diff
@@ -62,6 +71,9 @@ def _append(self, widget: Widget) -> None:
         if widget not in self._nodes_set:
             self._nodes.append(widget)
             self._nodes_set.add(widget)
+            widget_id = widget.id
+            if widget_id is not None and widget_id not in self._nodes_by_id:
```

Comment:
I think I see an issue.

Widgets can have duplicate IDs currently. If we add widget A and widget B with an id of "foo". Then widget A gets removed, I think that widget B will not be in the `_nodes_by_id` cache.

I think this means that `_nodes_by_id` should be a dict of lists.

Reply:
Nice one, very good catch